integration-tests/src/tests/runtime/sanity_checks.rs

```diff
@@ -0,0 +1,415 @@
+    assert_eq!(tx_result.status, FinalExecutionStatus::SuccessValue(to_base64(&[])));
+    assert_eq!(tx_result.receipts_outcome.len(), 1);
+
+    node
+}
+
+/// Calls method `sanity_check` on `test-contract-rs` and verifies that the
+/// resulting gas profile matches expectations.
+///
+/// This test intends to catch accidental configuration changes, see #4961.
+#[test]
+fn test_cost_sanity() {
+    let node = setup_runtime_node_with_contract(near_test_contracts::rs_contract());
+    let data = serde_json::json!({
+        "contract_code": to_base64(near_test_contracts::trivial_contract()),
+        "method_name": "main",
+        "method_args": to_base64(&[]),
+        "validator_id": bob_account().as_str(),
+    });
+    let res = node
+        .user()
+        .function_call(
+            alice_account(),
+            test_contract_account(),
+            "sanity_check",
+            serde_json::to_vec(&data).unwrap(),
+            MAX_GAS,
+            0,
+        )
+        .unwrap();
+    assert_eq!(res.status, FinalExecutionStatus::SuccessValue(to_base64(&[])));
+    assert_eq!(res.transaction_outcome.outcome.metadata.gas_profile, None);
+
+    let actual_statuses: Vec<ExecutionStatusView> =
+        res.receipts_outcome.iter().map(|outcome| outcome.outcome.status.clone()).collect();
+    let expected_statuses = expected_cost_sanity_receipts_statuses();
+    let mut diff_count = 0;
+    for i in 0..std::cmp::max(actual_statuses.len(), expected_statuses.len()) {
+        let actual = actual_statuses.get(i);
+        let expected = expected_statuses.get(i);
+        if actual != expected {
+            diff_count += 1;
+            println!("receipts[{}] has unexpected status:", i);
+            println!("\t+ {:?}", actual);
+            println!("\t- {:?}", expected);
+        }
+    }
+    if diff_count > 0 {
+        panic!("encountered {} receipt(s) with unexptected status", diff_count);
+    }
+
+    let actual_gas_profiles = res
+        .receipts_outcome
+        .iter()
+        .map(|outcome| outcome.outcome.metadata.gas_profile.as_ref().unwrap());
+    let expected_gas_profiles = expected_cost_sanity_gas_profiles();
+    assert_eq!(actual_gas_profiles.len(), expected_gas_profiles.len());
+    let mut diff_count = 0;
+    for (i, (actual_vec, expected_vec)) in
+        actual_gas_profiles.zip(expected_gas_profiles).enumerate()
+    {
+        for j in 0..std::cmp::max(actual_vec.len(), expected_vec.len()) {
+            let actual_cost = actual_vec.get(j);
+            let expected_cost = expected_vec.get(j);
+            if actual_cost != expected_cost {
+                diff_count += 1;
+                println!("unexpected cost at receipts[{}] gas_profile[{}]:", i, j);
+                println!("\t+ {:?}", actual_cost);
+                println!("\t- {:?}", expected_cost);
+            }
+        }
+    }
+    if diff_count > 0 {
+        panic!("encountered {} unexpected cost(s)", diff_count);
+    }
```

Comment:
Why not `assert_eq!(actual_gas_profiles, expected_gas_profiles);`? The current (more verbose) approach aims at making it easier to update `expected_gas_profiles` after costs changed.